tests/ops/qubit/test_parametric_ops.py

```diff
@@ -3112,9 +3112,294 @@ def test_empty_wire_list_error_multirz(self):
+            params = npp.array([[-50.0, 3.0, 50.0], [3.0, 50.0, -50.0], [50.0, -50.0, 3.0]])
+
+        # construct the wires
+        if op_class.num_wires == 1:
+            wires = 0
+        else:
+            wires = [0, 1]
+
+        return op_class(*params, wires)
+
+    @pytest.mark.parametrize("op", rotations)
+    def test_simplify_rotations(self, op):
+        """Test that the matrices and wires are the same after simplification"""
+
+        unsimplified_op = self.get_unsimplified_op(op)
+        simplified_op = qml.simplify(unsimplified_op)
+
+        assert qml.math.allclose(qml.matrix(unsimplified_op), qml.matrix(simplified_op))
+        assert all((p >= 0).all() and (p < 4 * np.pi).all() for p in simplified_op.data)
+        assert unsimplified_op.wires == simplified_op.wires
+
+    @pytest.mark.autograd
+    @pytest.mark.parametrize("op", rotations)
+    def test_simplify_rotations_grad_autograd(self, op):
+        """Test the gradient of an op after simplication for the autograd interface"""
+        dev = qml.device("default.qubit", wires=2)
+
+        @qml.qnode(dev)
+        def circuit(simplify, wires, *params):
+            if simplify:
+                qml.simplify(op(*params, wires))
+            else:
+                op(*params, wires)
+
+            return qml.expval(qml.PauliZ(0))
+
+        unsimplified_op = self.get_unsimplified_op(op)
+        params, wires = unsimplified_op.data, unsimplified_op.wires
+
+        params = [p[0] for p in params]
+
+        unsimplified_res = circuit(False, wires, *params)
+        simplified_res = circuit(True, wires, *params)
+
+        unsimplified_grad = qml.grad(circuit, argnum=list(range(2, 2 + len(params))))(
+            False, wires, *params
+        )
+        simplified_grad = qml.grad(circuit, argnum=list(range(2, 2 + len(params))))(
+            True, wires, *params
+        )
+
+        assert qml.math.allclose(unsimplified_res, simplified_res)
+        assert qml.math.allclose(unsimplified_grad, simplified_grad)
+
+    @pytest.mark.tf
+    @pytest.mark.parametrize("op", rotations)
+    def test_simplify_rotations_grad_tensorflow(self, op):
+        """Test the gradient of an op after simplication for the tensorflow interface"""
+        import tensorflow as tf
+
+        dev = qml.device("default.qubit", wires=2)
+
+        @qml.qnode(dev, interface="tf")
+        def circuit(simplify, wires, *params):
+            if simplify:
+                qml.simplify(op(*params, wires))
+            else:
+                op(*params, wires)
+
+            return qml.expval(qml.PauliZ(0))
+
+        unsimplified_op = self.get_unsimplified_op(op)
+        params, wires = unsimplified_op.data, unsimplified_op.wires
+
+        params = [tf.Variable(p[0]) for p in params]
+
+        with tf.GradientTape() as unsimplified_tape:
+            unsimplified_res = circuit(False, wires, *params)
+
+        unsimplified_grad = unsimplified_tape.gradient(unsimplified_res, params)
```

Comment:
can't we use `qml.grad(circ)(params)` here?

Reply:
`qml.grad` is just for the autograd interface.

Reply:
Indeed.